### PR TITLE
Settings page error validation

### PR DESCRIPTION
**Pull request by danachen**
## Description

Error validation from front end input, as well as backend checks:

1. User info form, you can test:
- Length of input for both user name and email address
- Both fields are filled
- Attempt to update email to an email already existing in DB is rejected

2. Password form, test:
- Old password is correct
- New passwords match
- Passwords are prompted to be of certain length

3. User language settings, test:
- Languages are not the same

Overall, check either success (green) or failure (red) messages show up for each submission attempt.

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|  x | :bug: Bug fix              |
|     | :sparkles: New feature     |
|  x | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

### CODE REVIEW

File: src/components/Settings.tsx [hunk 1]
```diff
@@ -138,14 +167,15 @@ export default function Settings() {
      <p className="text-sm mb-6 text-green-600 font-bold">{showUserMessage && usermessage}</p>
       <label className="label sr-only" htmlFor="username">Name</label>
         <input {...register('username', { required: true, minLength: 3, maxLength: 20 })} id="username" name="username" defaultValue={user.username} className="input appearance-none rounded-none relative block w-full px-3 py-2 border border-gray-300 placeholder-gray-500 text-gray-900 rounded-t-md focus:outline-none focus:ring-indigo-500 focus:border-indigo-500 focus:z-10 sm:text-sm" type="text" />
```
Comment (mrchrmn) [inline src/components/Settings.tsx:169]: I know two people called Bo. Maybe reduce the required length to 2? And why cap at 20?

File: src/components/Settings.tsx [hunk 2]
```diff
@@ -138,14 +167,15 @@ export default function Settings() {
      <p className="text-sm mb-6 text-green-600 font-bold">{showUserMessage && usermessage}</p>
       <label className="label sr-only" htmlFor="username">Name</label>
```
Comment (mrchrmn) [inline src/components/Settings.tsx:168]: For some reasons, the labels are not displayed. Please investigate.

File: src/components/Settings.tsx [hunk 3]
```diff
@@ -212,7 +239,7 @@ export default function Settings() {
       {<select {...register3('currentLearnLanguageId')} defaultValue={user?.learnLanguageId} className="input appearance-none rounded-none w-2/3 px-3 py-2 border border-gray-300 placeholder-gray-500 text-gray-900 rounded-b-md focus:outline-none focus:ring-indigo-500 focus:border-indigo-500 focus:z-10 sm:text-sm">
       {languages.map((lang) => <option key={lang.id} value={lang.id}>{lang.name} </option>)}
```
Comment (mrchrmn) [inline src/components/Settings.tsx:240]: You could capitalize the language names (use `capitalize` function from the navbar) and also include the flag (`lang.flag`).

File: src/components/Settings.tsx (hunk 2 above)
Comment (danachen) [reply src/components/Settings.tsx:168]: You mean "Name" and so on? Labels are not meant to be displayed (screen reader only).

File: src/components/Settings.tsx (hunk 1 above)
Comment (danachen) [reply src/components/Settings.tsx:169]: No reason, just seems unlikely. I can reduce the minLength requirement.

File: src/components/Settings.tsx (hunk 2 above)
Comment (mrchrmn) [reply src/components/Settings.tsx:168]: That is not correct, I think.
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label

The label for the language settings display just fine like I think they should. In any case, I would like to read on the page what each field is without relying on placeholders.

File: src/components/Settings.tsx (hunk 3 above)
Comment (danachen) [reply src/components/Settings.tsx:240]: > 

Will add those!

**Pull request merged**